### PR TITLE
RF: fix tests

### PR DESCRIPTION
**Pull request by agorshk**

### CODE REVIEW

File: daal4py/sklearn/ensemble/_forest.py [hunk 1]
```diff
@@ -704,32 +704,19 @@ def predict_proba(self, X):
...
-            if X.shape[1] != self.n_features_in_:
-                raise ValueError(
-                    (f'X has {X.shape[1]} features, '
-                     f'but RandomForestClassifier is expecting '
-                     f'{self.n_features_in_} features as input'))
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
+                        dtype=[np.float64, np.float32])
+        if not hasattr(self, 'daal_model_') or \
+                sp.issparse(X) or self.n_outputs_ != 1 or \
+                not daal_check_version((2021, 'P', 200)):
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/_forest.py:711]:
```suggestion

```

File: daal4py/sklearn/ensemble/_forest.py [hunk 2]
```diff
@@ -704,32 +704,19 @@ def predict_proba(self, X):
...
         """
-        # Temporary solution
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
-        if hasattr(self, 'n_features_in_'):
-            if X.shape[1] != self.n_features_in_:
-                raise ValueError(
-                    (f'X has {X.shape[1]} features, '
-                     f'but RandomForestClassifier is expecting '
-                     f'{self.n_features_in_} features as input'))
+        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/_forest.py:707]: move this after check for use daal, otherwise will do unnecessary checks

File: sklearnex/ensemble/tests/test_random_forest.py [hunk 3]
```diff
@@ -36,4 +36,4 @@ def test_sklearnex_import_rf_regression():
                            random_state=0, shuffle=False)
     rf = RandomForestRegressor(max_depth=2, random_state=0).fit(X, y)
     assert 'daal4py' in rf.__module__
-    assert_allclose([-6.66], rf.predict([[0, 0, 0, 0]]), atol=1e-2)
+    assert_allclose([-6.97], rf.predict([[0, 0, 0, 0]]), atol=1e-2)
```
Comment (PetrovKP) [inline sklearnex/ensemble/tests/test_random_forest.py:39]: what is the value in stock?

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 4]
```diff
@@ -31,9 +31,9 @@
     import RandomForestRegressor as DaalRandomForestRegressor
 from daal4py.sklearn._utils import daal_check_version
 
-ACCURACY_RATIO = 0.85
+ACCURACY_RATIO = 0.95 if daal_check_version((2021, 'P', 400)) else 0.85
```
Comment (PetrovKP) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:34]: I understand correctly, the accuracy has bad?

File: daal4py/sklearn/ensemble/_forest.py (hunk 2 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/_forest.py:707]: Ok

File: sklearnex/ensemble/tests/test_random_forest.py (hunk 3 above)
Comment (agorshk) [reply sklearnex/ensemble/tests/test_random_forest.py:39]: In stock Sklearn it's near -8,32; new implementation is close

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py (hunk 4 above)
Comment (agorshk) [reply daal4py/sklearn/ensemble/tests/test_decision_forest.py:34]: No, accuracy and log loss in new implementation is better

Comment (agorshk) [reply daal4py/sklearn/ensemble/tests/test_decision_forest.py:34]: Also I checked MSE in the regression example(prediction saves in file decision_forest_regression_batch.csv), it used to ~17,65; in new implementation it's ~17,26. It means that error is decreased.

**Comment by agorshk** [comment]
CI might be bound from this [PR](https://github.com/oneapi-src/oneDAL/pull/1725) We don't have to run CI twice

**Comment by agorshk** [comment]
/intelci: run
set repo=https://github.com/agorshk/daal.git
set branch=dev/agorshk-forest_feature_filtering

**Comment by agorshk** [comment]
/intelci: restart
set repo=https://github.com/agorshk/daal.git
set branch=dev/agorshk-forest_feature_filtering

**Comment by agorshk** [comment]
/intelci: restart
set ext_build=ebe5366e-233d-f1c1-a04c-8cdcd4b6d2c4

**Review by PetrovKP — APPROVED** [review]

File: deselected_tests.yaml [hunk 5]
```diff
@@ -235,6 +239,25 @@ gpu:
...
+  - ensemble/tests/test_voting.py::test_predict_proba_on_toy_problem
+  - ensemble/tests/test_voting.py::test_gridsearch
+  - ensemble/tests/test_voting.py::test_voting_classifier_set_params
+  - ensemble/tests/test_voting.py::test_set_estimator_drop
+  - ensemble/tests/test_voting.py::test_estimator_weights_format
+  - ensemble/tests/test_voting.py::test_transform
+  - inspection/tests/test_permutation_importance.py::test_permutation_importance_equivalence_sequential_parallel
+  - tests/test_calibration.py
+  - tests/test_common.py::test_estimators
+  - tests/test_multioutput.py::test_multi_output_classification
```
Comment (michael-smirnov) [inline deselected_tests.yaml:260]: Are this tests fail just because of predict_proba usage?
Is it clear what is broken in oneDAL?
I'm afraid that we broke big amount of tests that were passing previously.

**Review by michael-smirnov — CHANGES_REQUESTED** [review]

**Review by michael-smirnov — APPROVED** [review]

**Comment by agorshk** [comment]
@Mergifyio rebase
